source/common/http/conn_manager_impl.h

```diff
@@ -396,6 +402,7 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     // By default, we will assume there are no 100-Continue headers. If encode100ContinueHeaders
     // is ever called, this is set to true so commonContinue resumes processing the 100-Continue.
     bool has_continue_headers_{};
+    bool is_head_request_{false};
```

Comment:
do we need this at both levels for early response before we have filters?

Reply:
A head request only consider early response this scenario, with filters is upstream service is needed to deal with